components/TextFields/src/ContainedInputView/private/MDCContainedInputView.h

```diff
@@ -0,0 +1,107 @@
+// Copyright 2019-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <Foundation/Foundation.h>
+#import <UIKit/UIKit.h>
+
+#import "MDCContainedInputViewColorViewModel.h"
+#import "MDCContainedInputViewLabelAnimation.h"
+#import "MDCContainedInputViewLabelState.h"
+#import "MDCContainerStyleVerticalPositioningReference.h"
+#import "MDCTextControlLabelBehavior.h"
+#import "MDCTextControlState.h"
+
+static const CGFloat kMDCContainedInputViewDefaultAnimationDuration = (CGFloat)0.15;
+
+@protocol MDCContainedInputViewStyle;
+
+@protocol MDCContainedInputView <NSObject>
+
+/**
+ Dictates the @c MDCContainedInputViewStyle of the text field. Defaults to an instance of
+ MDCContainedInputViewStyleBase.
+ */
+@property(nonatomic, strong, nonnull) id<MDCContainedInputViewStyle> containerStyle;
+
+/**
+ Describes the current @c MDCtextControlState of the view.
+ */
+@property(nonatomic, assign, readonly) MDCTextControlState textControlState;
+
+/**
+ Describes the current @c MDCContainedInputViewLabelState of the contained input view. This
+ value is affected by things like the view's state, the value for @c canFloatingLabelFloat, and the
```

Comment:
where is the property canFloatingLabelFloat defined? i don't see it here or in the PR.

Reply:
Oof, I need to audit all these docs. Thank you for pointing this out!